index.js

```diff
@@ -78,12 +79,18 @@ globals.setImmediate(function () {
 	}
 
 	send('stats', {
-		testCount: numberOfTests
+		testCount: numberOfTests,
+		hasExclusive: hasExclusive
 	});
 
 	runner.on('test', test);
 
-	process.on('ava-run', function () {
+	process.on('ava-run', function (options) {
+		if (options.runOnlyExclusive && !hasExclusive) {
+			exit();
```

Comment:
Maybe pass `runOnlyExclusive` to `runner.run()`? It can still create its `stats` object inside the `run()` method, but bail out if it has no exclusive tests.